### DISCUSSION TITLE
Open hooked files

### DISCUSSION THREAD

**Discussion by moffat** [ttscoff/bunch-docs#224] [Ideas]
Hi,

I'm not sure how many Hook users use Bunch, but one feature that would be great to have in Bunch is the ability to open a file **and all hooked files to that file**. This would allow to set up a Bunch with a file of interest and then dynamically set (without having to edit the bunch every time) which files to open. I probably can do this with the scripting support in Bunch, but then one would lose the ability to close the files opened by a Bunch, I think...

All the best.

**Comment by ttscoff** [comment]
I don't hate this idea. Do you have a suggested syntax you would use? 
And are hooked files opened in their default application, or is this 
applied to a specific app? Does the original file always open, or only 
the files hooked to it? Should you have to specify the file once to open 
it, and then again for all hooked files? Open to suggestions.

```
Default
- (hooked:~/Documents/myfile.txt)
```

**Comment by moffat** [comment]
@ttscoff I take this as a huge compliment :)

For my use case, it seems like opening the file and hooked files at once is the simpler way to go. I like the (hooked:FILENAME) structure.

Now, would this mean that the hooked files are closed when the bunch is closed? I sure hope so!

Thanks, Brett.

**Comment by ttscoff** [comment]
Oh, no, Bunch has no ability to close individual files, only apps. You 
can execute a ⌘W, but which file closes is entirely a matter of what 
window is in the foreground. There's no way to track individual files 
after they've opened, at least not a way that works across all apps.

-Brett

**Reply by moffat** [reply]
Hi @ttscoff 

You're right of course. Perhaps if Hook added an option to close all hooked files sometime in the future?

Regardless, adding this feature to Bunch as you describe it above would be great. Thanks for considering it.

**Comment by moffat** [comment]
I realized I didn't answer this:

> And are hooked files opened in their default application, or is this applied to a specific app?

I'm not sure what you mean with the second option. I would think only the first is possible?
